### PR TITLE
chore: make teleoperators module pass type checks

### PR DESCRIPTION
**Pull request by Huang-JiaShun**
## What this does
This work addresses #1726 , which is part of #1719 .

## Related issues

- Closes: #1726 
- Related: #1719 

## Changes
Verified with mypy:  teleoperators module now passes cleanly.

Example:

- Ran the relevant tests:

  ```bash
  pytest -q tests/ -k <keyword>
  ```

- Reproduce with a quick example or CLI (if applicable):

  ```bash
  lerobot-train --some.option=true
  ```

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [x] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by imstevenpmwork** [comment]
Hello @Huang-JiaShun thanks for this contribution! We very recently added a new teleoperator and I merge `main` to this branch, so as expected mypy is now falling because we haven't checked this new teleoperator. Could you take a look? Otherwise I can do it in the following days

File: src/lerobot/teleoperators/gamepad/teleop_gamepad.py [hunk 1]
```diff
@@ -89,6 +90,7 @@ def connect(self) -> None:
     @check_if_not_connected
     def get_action(self) -> RobotAction:
         # Update the controller to get fresh inputs
+        assert self.gamepad is not None
```
Comment (imstevenpmwork) [inline src/lerobot/teleoperators/gamepad/teleop_gamepad.py]: could we use an `if` statement ? `assert` in src code is not ideal

**Comment by Huang-JiaShun** [comment]
"Hi! @imstevenpmwork Thanks for merging. I'll take a look at the mypy errors regarding the new teleoperator and push a fix shortly."
